qml/Stage/ApplicationWindow.qml

```diff
@@ -251,6 +259,7 @@ FocusScope {
                 name: "splashScreen"
                 when:
                      !d.hadSurface && (!root.surface || !d.surfaceInitialized)
+                     && (d.liveSurface ? d.liveSurface : !d.surfaceSeenLive)
```

Comment:
Isn't this equivalent to 
```suggestion
                     && (d.liveSurface || !d.surfaceSeenLive)
```

Reply:
My mind was pretty confused on correct condition when I tried to debug this.